components/server/ee/src/prebuilds/github-app.ts

```diff
@@ -103,6 +107,23 @@ export class GithubApp {
         app.on(['pull_request.opened', 'pull_request.synchronize', 'pull_request.reopened'], async ctx => {
             await this.handlePullRequest(ctx);
         });
+
+        options.getRouter && options.getRouter('/reconfigure').get('/', async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+            const gh = await app.auth();
+            const data = await gh.apps.getAuthenticated();
+            const slug = data.data.slug;
+
+            const state = req.query.state;
+            res.redirect(`https://github.com/apps/${slug}/installations/new?state=${state}`)
+        });
+        options.getRouter && options.getRouter('/setup').get('/', async (req: express.Request, res: express.Response, next: express.NextFunction) => {
```

Comment:
todo: re-introduce `req.query.state` which will be passed on `/reconfigure` in order to identify the origins of this request. IF we get called here without `state`, we can assume, that this is call from the marketplace, thus redirecting to `/install-github-app` should be done.